app/persona/controllers/Offers.scala

```diff
@@ -3,38 +3,65 @@ package persona.controllers
 import java.util.UUID
 import javax.inject.{Inject, Singleton}
 
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
 import persona.api.authentication.User
 import persona.api.offer.OfferService
 import play.api.libs.concurrent.Execution.Implicits._
 import play.api.libs.json._
 import play.api.mvc._
+import persona.api.offer.offerImpl._
 
 import scala.concurrent.Future
 
 @Singleton
-class Offers @Inject() (offerService: OfferService) extends Controller {
+class Offers @Inject() (offerService: OfferService,
+                         jsonOfferWriter: JsonOfferWriter) extends Controller {
 
   def list = Action.async {
-    offerService.list map { option =>
-      option map { offers =>
-        Ok(JsString("Listing offers!"))
-      } getOrElse {
-        InternalServerError
+    val offers = offerService.list
+      offers map {offer =>
+        val jsonOffers = offer.map(jsonOfferWriter.toJson)
+        val serializedJson = Json.toJson(jsonOffers)
+
+        Ok(serializedJson)
       }
-    }
   }
 
-  def get(id: Long) = Action.async {
-    offerService.get(id) map { option =>
-      option map { offer =>
-        Ok(JsString("Listing offer " + offer.id))
+//
+//def get(id: UUID) = Action.async {
+//
+//  offerService.get(id).map { optionOffer =>
+//    optionOffer map { offer =>
+//      val jsonOffers = optionOffer.map(jsonOfferWriter.toJson)
+//      val serializedJson = Json.toJson(jsonOffers)
+//
+//      Ok(serializedJson)
+//    } getOrElse {
+//      InternalServerError
+//    }
+//  }
+//}
+
+
+  def get(id: UUID) = Action.async {
+
+    offerService.get(id).map { optionOffer =>
+      optionOffer map { offer =>
+        val jsonOffers = optionOffer.map(jsonOfferWriter.toJson)
+        val serializedJson = Json.toJson(jsonOffers)
+
+        Ok(serializedJson)
       } getOrElse {
         InternalServerError
```

Comment:
404 Not found might be better than InternalServerError.  Or return 200 OK with an error message.  Should do some quick research and talking with @d77kim to determine which approach is more appropriate (for handling by clients)